### ISSUE TITLE
Using attributes in docstrings trips differing-param-doc

### ISSUE THREAD

**Issue by mitar** [PyCQA/pylint#1867]
### Steps to reproduce

I have the following class definition with docstrings in numpy style:

```python
class Foobar:
    """
    Parameters
    ----------
    foo : str
        Something.

    Attributes
    ----------
    bar : str
        Something.
    """

    def __init__(self, foo):
        self.bar = None
```

### Current behavior

When I run pylint I get:

```
W:  1, 0: "bar" differing in parameter documentation (differing-param-doc)
W:  1, 0: "bar" differing in parameter type documentation (differing-type-doc)
```

### Expected behavior

No warning to be issued. Attributes are not parameters.

### pylint --version output

```
pylint 1.8.0, 
astroid 1.6.1
Python 3.6.3 (default, Jan 25 2018, 23:14:44) 
[GCC 5.4.0 20160609]
```

### Workaround

If I swap `Parameters` and `Attributes` sections it works. It seems that `Attributes` heading is simply ignored and all entries under `Attributes` are taken as parameters.

**Comment by PCManticore** [comment]
cc @AWhetter

**Comment by AWhetter** [comment]
I've just had a quick look at this. I'm not 100% sure but I think that it isn't detecting the end of the section, and so it counts `bar` as part of the parameters section and errors.
So swapping the sections around so that `Parameters` comes after `Attributes` would also not raise the warning?

**Comment by mitar** [comment]
> So swapping the sections around so that Parameters comes after Attributes would also not raise the warning?

Yes, swapping sections fixes the problem (warning is not shown). I think my understanding what is the issue is the same as yours. This is why workaround works.
